leasing/serializers/rent.py

```diff
@@ -578,61 +574,182 @@ def validate_seasonal_values(self, rent_data: dict) -> None:
         validate_seasonal_day_for_month(start_day, start_month)
         validate_seasonal_day_for_month(end_day, end_month)
 
-    def validate_override_receivable_type_value(self, rent_data: dict) -> None:
+    def validate_override_receivable_type_value(
+        self, rent_data: dict[str, Any]
+    ) -> None:
         """
-        Override receivabletype is mandatory for AKV and KuVa leases, and not
-        used by MaKe/Tontit.
+        Currently, the rent override receivabletype is mandatory for AKV and
+        KuVa leases, and not used by MaKe/Tontit.
 
         It is only used in index rents, fixed rents, and manual rents, because
         these rent types can generate automatic invoices.
 
+        TODO Blind spot: this implementation cannot reject missing override
+        receivable type when creating a new rent, because the validator doesn't
+        receive enough details from the frontend.
+        - Cannot read service unit from the receivable type if it's missing
+        - Cannot find out the service unit from rent's lease if rent
+        doesn't exist yet.
+
         Raises: serializers.ValidationError
         """
-        override_receivable_type = rent_data.get("override_receivable_type")
+        rent_type = rent_data.get("type", {})
+        rent_types_that_can_use_override_receivable_type = [
+            RentType.INDEX,
+            RentType.INDEX2022,
+            RentType.FIXED,
+            RentType.MANUAL,
+        ]
+        # These rent types can generate automatic invoices, so they can utilize
+        # the override receivabletype, if required by service unit.
+        rent_type_uses_override = (
+            rent_type in rent_types_that_can_use_override_receivable_type
+        )
 
-        if not override_receivable_type:
-            # Empty override receivabletype input should be allowed, because it
-            # is always empty for all MaKe rents, and those must be allowed.
+        rent_id = rent_data.get("id")
+        if rent_id is None:
+            # The ID is not present in creation flow, and is removed from
+            # rent_data before later validations also during the update flow.
+            # Try get ID from the rent instance, which is present if an existing
+            # rent is being updated.
+            rent_id = self.instance.id if self.instance else None
```

Comment:
Just for info, another way of accomplishing this could be:
`getattr(self.instance, "id", None)`

Reply:
Right, forgot about that one